### ISSUE TITLE
Top level "hosts" section probably should have another name

### ISSUE THREAD

**Issue by morrone** [ovis-hpc/maestro#4]
@tom95858 I think that the name "hosts" is used in two very different ways in the current maestro yaml, and that should probably be changes. The top-level "hosts" section should probably be renamed something like "daemons", because as far as I can tell it actually is used to configure where ldmsd daemons live, not to configure hosts. And in fact the example shows multiple of these "hosts" living on the same "host".

**Comment by tom95858** [comment]
I went around and around on this. At the end of the day, I came to the conclusion that a host is a listening endpoint. But perhaps we need to revisit (rename) this. Listening endpoints need to be configured, grouped and monitored. The grouping aspect is so that aggregators (also listeners) can be configured to gather data from some pool of data providers (samplers, and other aggregators). I wanted it to be easy to express which 'hosts' are being monitored by which other 'hosts'. 

It may be that this current format is prejudiced by the way we do things in the server as opposed to the relationships that really need to be expressed. I thought about adding a 'role' section to each 'host', but I found that the configuration became highly nested and I didn't like that. So the 'role' data is broken out into different sections and 'hosts' are applied to those roles in the host-list.
